blobs/howto/python/blob-devguide-py/blob-devguide-blobs.py

```diff
@@ -0,0 +1,410 @@
+        blob_client = blob_service_client.get_blob_client(container=container_name, blob="sample-blob.txt")
+
+        # Get the existing blob properties
+        properties = blob_client.get_blob_properties()
+
+        # Set the content_type and content_language headers, and populate the remaining headers from the existing properties
+        blob_headers = ContentSettings(content_type="text/plain",
+                                       content_encoding=properties.content_settings.content_encoding,
+                                       content_language="en-US",
+                                       content_disposition=properties.content_settings.content_disposition,
+                                       cache_control=properties.content_settings.cache_control,
+                                       content_md5=properties.content_settings.content_md5)
+        
+        blob_client.set_http_headers(blob_headers)
+    # </Snippet_set_blob_properties>
+
+    # <Snippet_set_blob_metadata>
+    def set_metadata(self, blob_service_client: BlobServiceClient, container_name):
+        blob_client = blob_service_client.get_blob_client(container=container_name, blob="sample-blob.txt")
+
+        # Retrieve existing metadata, if desired
+        metadata = dict(blob_client.get_blob_properties().metadata)
+
+        more_metadata = {'docType': 'text', 'docCategory': 'reference'}
+        metadata.update(more_metadata)
+
+        # Set metadata on the blob
+        blob_client.set_blob_metadata(metadata=metadata)
+    # </Snippet_set_blob_metadata>
+
+    # <Snippet_get_blob_metadata>
+    def get_metadata(self, blob_service_client: BlobServiceClient, container_name):
+        blob_client = blob_service_client.get_blob_client(container=container_name, blob="sample-blob.txt")
+
+        # Retrieve existing metadata, if desired
+        metadata = dict(blob_client.get_blob_properties().metadata)
+
+        for k, v in metadata.items():
+            print(k, v)
+    # </Snippet_get_blob_metadata>
+
+    # <Snippet_list_blobs_flat>
+    def list_blobs_flat(self, blob_service_client: BlobServiceClient, container_name):
+        container_client = blob_service_client.get_container_client(container=container_name)
+
+        blob_list = container_client.list_blobs()
+
+        for blob in blob_list:
+            print(f"Name: {blob.name}")
+    # </Snippet_list_blobs_flat>
+
+    # <Snippet_list_blobs_flat_options>
+    def list_blobs_flat_options(self, blob_service_client: BlobServiceClient, container_name):
+        container_client = blob_service_client.get_container_client(container=container_name)
+
+        blob_list = container_client.list_blobs(include=['tags'])
+
+        for blob in blob_list:
+            print(f"Name: {blob['name']}, Tags: {blob['tags']}")
+    # </Snippet_list_blobs_flat_options> 
+
+    # <Snippet_list_blobs_hierarchical>
+    depth = 0
+    indent = "  "
```

Comment:
Probably should move these inside the sample function rather than defining them globally?